src/Microsoft.Azure.ServiceBus/MessageSender.cs

```diff
@@ -35,18 +35,55 @@ public Task SendAsync(IEnumerable<BrokeredMessage> brokeredMessages)
             return this.OnSendAsync(brokeredMessages);
         }
 
+        public Task<long> ScheduleMessageAsync(BrokeredMessage message, DateTimeOffset scheduleEnqueueTimeUtc)
+        {
+            if (message == null)
+            {
+                throw Fx.Exception.ArgumentNull(nameof(message));
+            }
+
+            if (scheduleEnqueueTimeUtc.CompareTo(DateTimeOffset.UtcNow) < 0)
+            {
+                throw Fx.Exception.ArgumentOutOfRange(
+                    nameof(scheduleEnqueueTimeUtc),
+                    scheduleEnqueueTimeUtc.ToString(),
+                    "Cannot schedule messages in the past");
+            }
+
+            message.ScheduledEnqueueTimeUtc = scheduleEnqueueTimeUtc.UtcDateTime;
+            MessageSender.ValidateMessage(message);
+            return this.OnScheduleMessageAsync(message);
+        }
+
+        public Task CancelScheduledMessageAsync(long sequenceNumber)
+        {
+            return this.OnCancelScheduledMessageAsync(sequenceNumber);
+        }
+
         protected abstract Task OnSendAsync(IEnumerable<BrokeredMessage> brokeredMessages);
 
+        protected abstract Task<long> OnScheduleMessageAsync(BrokeredMessage brokeredMessage);
+
+        protected abstract Task OnCancelScheduledMessageAsync(long sequenceNumber);
+
         static void ValidateMessages(IEnumerable<BrokeredMessage> brokeredMessages)
         {
             if (brokeredMessages == null || !brokeredMessages.Any())
             {
                 throw Fx.Exception.ArgumentNull(nameof(brokeredMessages));
             }
 
-            if (brokeredMessages.Any(brokeredMessage => brokeredMessage.IsLockTokenSet))
+            foreach (var brokeredMessage in brokeredMessages)
+            {
+                ValidateMessage(brokeredMessage);
+            }
+        }
+
+        static void ValidateMessage(BrokeredMessage brokeredMessage)
+        {
+            if (brokeredMessage.IsLockTokenSet)
             {
-                throw Fx.Exception.Argument(nameof(brokeredMessages), "Cannot Send ReceivedMessages");
+                throw Fx.Exception.Argument(nameof(brokeredMessage), "Cannot Send ReceivedMessages");
```

Comment:
The `validateMessage()` is being called only for `SendMessage` and `ScheduleMessage` which are pretty much same.